subsys/bluetooth/controller/ll_sw/ull_sync.c

```diff
@@ -108,11 +108,30 @@ uint8_t ll_sync_create(uint8_t options, uint8_t sid, uint8_t adv_addr_type,
 		return BT_HCI_ERR_MEM_CAPACITY_EXCEEDED;
 	}
 
+	scan->per_scan.state = LL_SYNC_STATE_IDLE;
+	node_rx->link = link_sync_estab;
+	scan->per_scan.node_rx_estab = node_rx;
 	scan->per_scan.filter_policy = options & BIT(0);
+	if (IS_ENABLED(CONFIG_BT_CTLR_PHY_CODED)) {
+		scan_coded->per_scan.state = LL_SYNC_STATE_IDLE;
+		scan_coded->per_scan.node_rx_estab =
+			scan->per_scan.node_rx_estab;
+		scan_coded->per_scan.filter_policy =
+			scan->per_scan.filter_policy;
+	}
+
 	if (!scan->per_scan.filter_policy) {
 		scan->per_scan.sid = sid;
 		scan->per_scan.adv_addr_type = adv_addr_type;
 		memcpy(scan->per_scan.adv_addr, adv_addr, BDADDR_SIZE);
+
+		if (IS_ENABLED(CONFIG_BT_CTLR_PHY_CODED)) {
+			scan_coded->per_scan.sid = scan->per_scan.sid;
+			scan_coded->per_scan.adv_addr_type =
+				scan->per_scan.adv_addr_type;
+			memcpy(scan_coded->per_scan.adv_addr,
+			       scan->per_scan.adv_addr, BDADDR_SIZE);
+		}
```

Comment:
It looks like, and correct me if I'm wrong, that `scan->per_scan` and `scan_coded->per_scan`  are identical after this initialization. 

Would it make sense to keep only the initialization for `scan` and then do 
```
memcpy(&scan_coded->per_scan, scan->per_scan, sizeof(scan_coded->per_scan);
```

rather than having duplicating code? 

Or perhaps have a init function to initialize them equally, e.g. `per_scan_init(&scan)` and `per_scan_init(&scan_coded)`?

Reply:
Actually, on doing a review of the `struct ll_scan_set` I see most of the members can be moved out as a shared struct, and should save memory too. I will get back to optimizing it once I get the 251 byte PDU support in place.